projects/ngx-d3timeline/src/lib/ngx-d3timeline.service.ts

```diff
@@ -36,8 +49,14 @@ export class NgxD3TimelineService implements OnDestroy {
   resourceTickMarkRectangles$ = this.store.select(selectResourceTickRectangles);
 
   private destroySubject = new Subject<boolean>();
-
-  constructor(private store: Store, private axisService: AxisService) {}
+  constructor(
+    private store: Store,
+    private axisService: AxisService,
+    hostElement: ElementRef,
```

Comment:
this will have to be made a field `private hostElement: ElementRef` to move `this.setupResizing` into the onInit of constructor.

Reply:
So are you saying inject the elementref and zone here, then call setupResizing with no parameters from the components onInit() ?

Reply:
Yeah, just because a constructor should really only be used to assign to fields. Any more work than that is better off in lifecycle hooks

Reply:
done